pkgs/development/libraries/pipewire/default.nix

```diff
@@ -189,6 +189,7 @@ let
       "-Dsdl2=disabled" # required only to build examples, causes dependency loop
       "-Drlimits-install=false" # installs to /etc, we won't use this anyway
       "-Dcompress-offload=enabled"
+      "-Dsystemd-system-unit-dir=${placeholder "out"}/lib/systemd"
```

Comment:
[This change](https://github.com/systemd/systemd/commit/b0d3095fd6cc1791a38f57a1982116b4475244ba#diff-3dce99316f44198c53d8147020aaca9c8931b5248587c64c098fdcef8ed4da03) appears to cause a regression in overridability, which we previously restored in https://github.com/systemd/systemd/commit/60bce7c6d9606185114df1bdcd5ea100407688b8.

Many projects have been overriding `root_prefix` and will now be broken:
https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/blob/e35655b364d32fce787cd3980c876f3946f82192/meson.build#L204

Reply:
Yes, we were discussing this on matrix today. Ideally, we would fix every package in the world to handle this correctly, instead of in the flawed way they do it now with `root(_)prefix`. But it's simpler to just make systemd do things the weird bad way:

```diff
diff --git a/src/core/systemd.pc.in b/src/core/systemd.pc.in
index 6cca2fad9a..8e39c5247e 100644
--- a/src/core/systemd.pc.in
+++ b/src/core/systemd.pc.in
@@ -11,9 +11,9 @@
 # considered deprecated (though there is no plan to remove them). New names
 # shall have underscores.
 
-prefix={{PREFIX_NOSLASH}}
-root_prefix=${prefix}
-rootprefix=${prefix}
+root_prefix={{PREFIX_NOSLASH}}
+rootprefix=${root_prefix}
+prefix=${rootprefix}
 sysconf_dir={{SYSCONF_DIR}}
 sysconfdir=${sysconf_dir}
```

This patch seems to be fixing builds for me, though it's really hard to test because my personal hydra is really unhappy doing world-rebuilds like this and I'm finding all sorts of nondeterministic build failures that are getting in the way of testing that this actually works -_-

Reply:
Well, `--define-variable` is [standard part of pkg-config](https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/) so I would argue https://github.com/systemd/systemd/commit/b0d3095fd6cc1791a38f57a1982116b4475244ba broke a public API.

But in the end, it depends on systemd’s policy – maybe they do not consider this part of their API promise. Though, they [accepted our fix](https://github.com/systemd/systemd/pull/18118) when they broke it in the past.

Personally, I would prefer applying your patch upstream and announcing the `root_prefix` for pkg-config override is deprecated, so that we have some time to fix it in projects installing service files.

Unfortunately, meson currently [does not support multiple `define_variable`](https://github.com/mesonbuild/meson/pull/10670) so projects will need to do something like this:

https://github.com/fwupd/fwupd/blob/b678ebeb0ae57c9b9c335db262996c4a8e8ff5c4/meson.build#L465-L469

Reply:
AFAIU this is fixed by https://github.com/systemd/systemd/pull/30115 in 255-rc3